sql_ddl/db.sql

```diff
@@ -2,53 +2,132 @@ create type user_role as enum ('admin', 'manager', 'engineer');
 create type user_status as enum ('active', 'inactive');
 create type action_status as enum ('active' , 'inactive');
 create type compound_status as enum ('active' , 'inactive');
+create type project_status as enum ('active' , 'inactive');
+create type test_case_status as enum ('active' , 'inactive');
+create type test_scenario_status as enum ('active' , 'inactive');
+create type test_case_result as enum ('failed' , 'complete');
+create type user_project_status as enum ('watcher' , 'developer');
+
+create table reset_token
+(
+    id           serial primary key,
+    token        varchar(100),
+    expiry_date  timestamp,
+    user_id integer UNIQUE NOT NULL REFERENCES users(id)
+);
+
 create table users (
                        id              serial primary key not null ,
-                       email           varchar(100) not null ,
+                       email           varchar(100) UNIQUE not null ,
                        password        varchar(300) not null ,
                        name            varchar(100) not null ,
                        role            user_role not null ,
                        status          user_status not null ,
-                       reset_token_id  integer,
                        last_request  timestamp
 );
 
-create table reset_token
+create table test_case
 (
-    id           serial primary key,
-    token        varchar(50),
-    expiry_date  timestamp,
-    user_id integer
+    id serial PRIMARY KEY NOT NULL,
+    project_id integer REFERENCES project(id),
+    status test_case_status NOT NULL,
+    result test_case_result NOT NULL
+);
+
+CREATE TABLE data_set
+(
+    id serial PRIMARY KEY NOT NULL,
+    title varchar(100),
+    description varchar(300),
+    test_case_id integer NOT NULL REFERENCES test_case(id)
```

Comment:
in er diagram it was id_test_case. I think names must be the same as in er diagram to avoid missunderstanding.